common/ManagementCoreShared/SharedExtensions.cs

```diff
@@ -2,15 +2,16 @@
 // Licensed under the MIT License.
 
 using System;
-using System.ComponentModel;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
 
-namespace Azure.ResourceManager
+namespace Azure.ResourceManager.Core
 {
     /// <summary>
-    /// Extensions over ResourceIdentifier types.
+    /// helper class
     /// </summary>
-    [EditorBrowsable(EditorBrowsableState.Never)]
-    public static class ResourceIdentifierExtensions
+    internal static class SharedExtensions
```

Comment:
public to internal? So AppendProviderResource and AppendChildResource are now hidden?

Reply:
hidden but shared, they will be included as shared code in all mgmt plane sdk

Reply:
I think these are useful methods for enduser of ResourceIdentifier not just internal logic.

Reply:
yes they did, and we had editor browsable never for the same reason.  This is just a better way to hide it.

I think we can add these kinds of helpers later but for preview we want to keep the api surface as small as possible

Reply:
Does this mean that the customers are not encouraged to use these APIs?